test/local/mysql_tests.js

```diff
@@ -221,6 +221,109 @@ DB.connect(config)
       )
 
       test(
+        'db.countAccountsCreatedBefore(Date.now())',
+        function (t) {
+          t.plan(1)
+          return db.countAccountsCreatedBefore(Date.now())
+          .then(function(result) {
+            t.ok(result.count >= 0, 'returned non-negative count')
+          }, function() {
+            t.fail('should have succeeded')
+          })
+        }
+      )
+
+      test(
+        'db.countAccountsCreatedBefore(0)',
+        function (t) {
+          return db.countAccountsCreatedBefore(0)
+          .then(function(result) {
+            // Count should be zero if STRICT_ALL_TABLES and
+            // STRICT_TRANS_TABLES are not set in my.cnf
+            t.plan(1)
+            t.equal(result.count, 0)
+          }, function(err) {
+            // Query should fail if STRICT_ALL_TABLES or
+            // STRICT_TRANS_TABLES are set in my.cnf
```

Comment:
The alternative here is to swallow the MySQL error in the wrapping functions and return a count of zero instead. But that feels like it might be a bit subversive.